Meta/run.py

```diff
@@ -0,0 +1,978 @@
+        # No native Windows QEMU, reconfigure to Linux QEMU without KVM
+        else:
+            config.virtualization_support = False
+            config.qemu_kind = QEMUKind.Other
+
+    if config.qemu_binary is None:
+        # Setup full path for the binary if possible (otherwise trust system PATH)
+        if os.path.exists(
+            local_qemu_bin := os.path.join(
+                "../Toolchain/Local/qemu/bin/", qemu_binary_basename
+            )
+        ):
+            config.qemu_binary = local_qemu_bin
+        if os.path.exists(
+            old_local_qemu_bin := os.path.join(
+                "../Toolchain/Local/x86_64/bin/", qemu_binary_basename
+            )
+        ):
+            config.qemu_binary = old_local_qemu_bin
+
+
+def check_qemu_version(config: Configuration):
+    if config.qemu_binary is None:
+        raise RunError(
+            f"Please install QEMU version {QEMU_MINIMUM_REQUIRED_MAJOR_VERSION}.{QEMU_MINIMUM_REQUIRED_MINOR_VERSION} or newer or use the Toolchain/BuildQemu.sh script."  # noqa: E501
+        )
+    version_information = run(
+        [config.qemu_binary, "-version"], capture_output=True, encoding="utf-8"
+    ).stdout
+    qemu_version_regex = re.compile(
+        r"QEMU emulator version ([1-9][0-9]*|0)\.([1-9][0-9]*|0)"
+    )
+    version_groups = qemu_version_regex.search(version_information)
+    if version_groups is None:
+        raise RunError(
+            f'QEMU seems to be defective, its version information is "{version_information}"'
+        )
+    major, minor = int(version_groups.group(1)), int(version_groups.group(2))
+
+    if major < QEMU_MINIMUM_REQUIRED_MAJOR_VERSION or (
+        major == QEMU_MINIMUM_REQUIRED_MAJOR_VERSION
+        and minor < QEMU_MINIMUM_REQUIRED_MINOR_VERSION
+    ):
+        raise RunError(
+            f"Required QEMU >= {QEMU_MINIMUM_REQUIRED_MAJOR_VERSION}.{QEMU_MINIMUM_REQUIRED_MINOR_VERSION}! Found {major}.{minor}. Please install a newer version of QEMU or use the Toolchain/BuildQemu.sh script."  # noqa: E501
+        )
+
+
+def setup_virtualization_support(config: Configuration):
+    provided_virtualization_enable = environ.get("SERENITY_VIRTUALIZATION_SUPPORT")
+    # The user config always forces the platform-appropriate virtualizer to be used,
+    # even if we couldn't detect it otherwise; this is intended behavior.
+    if provided_virtualization_enable is not None:
+        config.virtualization_support = provided_virtualization_enable == "1"
+    else:
+        if (
+            config.kvm_usable
+            and config.architecture != Arch.Aarch64
+            and os.uname().machine != Arch.Aarch64.value
+        ) or (
+            config.qemu_kind == QEMUKind.NativeWindows
+            and config.architecture != Arch.Aarch64
+        ):
+            config.virtualization_support = True
+
+    if config.virtualization_support:
+        available_accelerators = run(
+            [cast(str, config.qemu_binary), "-accel", "help"],
```

Comment:
cast? What's wrong with `str()` ? qemu_binary should probably be a pathlib.Path anyway

Reply:
It makes the typechecker assume that this cannot be None anymore, which we know for sure but cannot be statically determined. But yes, it should be `pathlib.Path | None`